python/ray/tune/suggest/bayesopt.py

```diff
@@ -202,8 +202,11 @@ def _setup_optimizer(self):
         if self._analysis is not None:
             self.register_analysis(self._analysis)
 
-    def set_search_properties(self, metric: Optional[str], mode: Optional[str],
-                              config: Dict) -> bool:
+    def set_search_properties(self,
+                              metric: Optional[str],
+                              mode: Optional[str],
+                              config: Dict,
+                              experiment: Optional[Dict] = None) -> bool:
```

Comment:
For forwards compatibility, should we add `**spec` here, too? (and for the other searchers as well)

Reply:
Actually, shouldn't we use `**spec` here instead (and in all searchers except those actually using `experiment`)? Or alternatively use `experiment` everywhere